packages/core/helper-plugin/src/hooks/useQueryParams.ts

```diff
@@ -3,25 +3,27 @@ import { useCallback, useMemo } from 'react';
 import { parse, stringify } from 'qs';
 import { useHistory, useLocation } from 'react-router-dom';
 
-const useQueryParams = (initialParams) => {
+type JSON = string | number | boolean | null | { [key: string]: JSON } | Array<JSON>;
+
+const useQueryParams = <T extends Record<string, JSON>>(initialParams?: T) => {
   const { search } = useLocation();
   const { push } = useHistory();
 
   const query = useMemo(() => {
     const searchQuery = search.substring(1);
 
     if (!search) {
-      return initialParams;
+      return initialParams || {};
     }
 
     return parse(searchQuery);
   }, [search, initialParams]);
 
   const setQuery = useCallback(
-    (nextParams, method = 'push') => {
+    (nextParams: T, method: 'push' | 'remove' = 'push') => {
       let nextQuery = { ...query };
 
-      if (method === 'remove') {
+      if (method === 'remove' && Object.keys(nextQuery).length !== 0) {
```

Comment:
How come you need to check there's keys length? Won't the below code just not loop through nothin if there are no keys?